// mediasoup/node/lib/fbs/transport/sctp-negotiated-max-streams-notification.js
"use strict";
// automatically generated by the FlatBuffers compiler, do not modify
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SctpNegotiatedMaxStreamsNotificationT = exports.SctpNegotiatedMaxStreamsNotification = void 0;
/* eslint-disable @typescript-eslint/no-unused-vars, @typescript-eslint/no-explicit-any, @typescript-eslint/no-non-null-assertion */
const flatbuffers = __importStar(require("flatbuffers"));
class SctpNegotiatedMaxStreamsNotification {
    bb = null;
    bb_pos = 0;
    __init(i, bb) {
        this.bb_pos = i;
        this.bb = bb;
        return this;
    }
    static getRootAsSctpNegotiatedMaxStreamsNotification(bb, obj) {
        return (obj || new SctpNegotiatedMaxStreamsNotification()).__init(bb.readInt32(bb.position()) + bb.position(), bb);
    }
    static getSizePrefixedRootAsSctpNegotiatedMaxStreamsNotification(bb, obj) {
        bb.setPosition(bb.position() + flatbuffers.SIZE_PREFIX_LENGTH);
        return (obj || new SctpNegotiatedMaxStreamsNotification()).__init(bb.readInt32(bb.position()) + bb.position(), bb);
    }
    negotiatedMaxOutboundStreams() {
        const offset = this.bb.__offset(this.bb_pos, 4);
        return offset ? this.bb.readUint16(this.bb_pos + offset) : 0;
    }
    negotiatedMaxInboundStreams() {
        const offset = this.bb.__offset(this.bb_pos, 6);
        return offset ? this.bb.readUint16(this.bb_pos + offset) : 0;
    }
    static startSctpNegotiatedMaxStreamsNotification(builder) {
        builder.startObject(2);
    }
    static addNegotiatedMaxOutboundStreams(builder, negotiatedMaxOutboundStreams) {
        builder.addFieldInt16(0, negotiatedMaxOutboundStreams, 0);
    }
    static addNegotiatedMaxInboundStreams(builder, negotiatedMaxInboundStreams) {
        builder.addFieldInt16(1, negotiatedMaxInboundStreams, 0);
    }
    static endSctpNegotiatedMaxStreamsNotification(builder) {
        const offset = builder.endObject();
        return offset;
    }
    static createSctpNegotiatedMaxStreamsNotification(builder, negotiatedMaxOutboundStreams, negotiatedMaxInboundStreams) {
        SctpNegotiatedMaxStreamsNotification.startSctpNegotiatedMaxStreamsNotification(builder);
        SctpNegotiatedMaxStreamsNotification.addNegotiatedMaxOutboundStreams(builder, negotiatedMaxOutboundStreams);
        SctpNegotiatedMaxStreamsNotification.addNegotiatedMaxInboundStreams(builder, negotiatedMaxInboundStreams);
        return SctpNegotiatedMaxStreamsNotification.endSctpNegotiatedMaxStreamsNotification(builder);
    }
    unpack() {
        return new SctpNegotiatedMaxStreamsNotificationT(this.negotiatedMaxOutboundStreams(), this.negotiatedMaxInboundStreams());
    }
    unpackTo(_o) {
        _o.negotiatedMaxOutboundStreams = this.negotiatedMaxOutboundStreams();
        _o.negotiatedMaxInboundStreams = this.negotiatedMaxInboundStreams();
    }
}
exports.SctpNegotiatedMaxStreamsNotification = SctpNegotiatedMaxStreamsNotification;
class SctpNegotiatedMaxStreamsNotificationT {
    negotiatedMaxOutboundStreams;
    negotiatedMaxInboundStreams;
    constructor(negotiatedMaxOutboundStreams = 0, negotiatedMaxInboundStreams = 0) {
        this.negotiatedMaxOutboundStreams = negotiatedMaxOutboundStreams;
        this.negotiatedMaxInboundStreams = negotiatedMaxInboundStreams;
    }
    pack(builder) {
        return SctpNegotiatedMaxStreamsNotification.createSctpNegotiatedMaxStreamsNotification(builder, this.negotiatedMaxOutboundStreams, this.negotiatedMaxInboundStreams);
    }
}
exports.SctpNegotiatedMaxStreamsNotificationT = SctpNegotiatedMaxStreamsNotificationT;
