// mediasoup/node/lib/fbs/sctp-parameters/num-sctp-streams.js
"use strict";
// automatically generated by the FlatBuffers compiler, do not modify
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.NumSctpStreamsT = exports.NumSctpStreams = void 0;
/* eslint-disable @typescript-eslint/no-unused-vars, @typescript-eslint/no-explicit-any, @typescript-eslint/no-non-null-assertion */
const flatbuffers = __importStar(require("flatbuffers"));
class NumSctpStreams {
    bb = null;
    bb_pos = 0;
    __init(i, bb) {
        this.bb_pos = i;
        this.bb = bb;
        return this;
    }
    static getRootAsNumSctpStreams(bb, obj) {
        return (obj || new NumSctpStreams()).__init(bb.readInt32(bb.position()) + bb.position(), bb);
    }
    static getSizePrefixedRootAsNumSctpStreams(bb, obj) {
        bb.setPosition(bb.position() + flatbuffers.SIZE_PREFIX_LENGTH);
        return (obj || new NumSctpStreams()).__init(bb.readInt32(bb.position()) + bb.position(), bb);
    }
    os() {
        const offset = this.bb.__offset(this.bb_pos, 4);
        return offset ? this.bb.readUint16(this.bb_pos + offset) : 65535;
    }
    mis() {
        const offset = this.bb.__offset(this.bb_pos, 6);
        return offset ? this.bb.readUint16(this.bb_pos + offset) : 65535;
    }
    static startNumSctpStreams(builder) {
        builder.startObject(2);
    }
    static addOs(builder, os) {
        builder.addFieldInt16(0, os, 65535);
    }
    static addMis(builder, mis) {
        builder.addFieldInt16(1, mis, 65535);
    }
    static endNumSctpStreams(builder) {
        const offset = builder.endObject();
        return offset;
    }
    static createNumSctpStreams(builder, os, mis) {
        NumSctpStreams.startNumSctpStreams(builder);
        NumSctpStreams.addOs(builder, os);
        NumSctpStreams.addMis(builder, mis);
        return NumSctpStreams.endNumSctpStreams(builder);
    }
    unpack() {
        return new NumSctpStreamsT(this.os(), this.mis());
    }
    unpackTo(_o) {
        _o.os = this.os();
        _o.mis = this.mis();
    }
}
exports.NumSctpStreams = NumSctpStreams;
class NumSctpStreamsT {
    os;
    mis;
    constructor(os = 65535, mis = 65535) {
        this.os = os;
        this.mis = mis;
    }
    pack(builder) {
        return NumSctpStreams.createNumSctpStreams(builder, this.os, this.mis);
    }
}
exports.NumSctpStreamsT = NumSctpStreamsT;
